scrapy/linkextractors/lxmlhtml.py

```diff
@@ -181,10 +183,29 @@ def __init__(
         strip: bool = True,
         restrict_text: _RegexOrSeveralT | None = None,
     ):
-        tags, attrs = set(arg_to_iter(tags)), set(arg_to_iter(attrs))
+        use_all_tags = tags is True
+        use_all_attrs = attrs is True
+
+        tags_set = set() if use_all_tags else set(arg_to_iter(tags))
+        attrs_set = set() if use_all_attrs else set(arg_to_iter(attrs))
+        deny_tags_set = set(arg_to_iter(deny_tags))
+        deny_attrs_set = set(arg_to_iter(deny_attrs))
+
+        def tag_checker(tag):
```

Comment:
Excellent changes. As a suggestion, you could consolidate the tag_checker into a single reusable function to reduce redundancy and improve maintainability.

```suggestion
     def create_checker(allowed, denied):
         if allowed is True:
             return lambda x: x not in denied
         if denied is True:
             return lambda x: x in allowed
         return partial(operator.contains, allowed)

     tag_checker = create_checker(tags_set, deny_tags_set)
     attr_checker = create_checker(attrs_set, deny_attrs_set)
```

Reply:
Thank you for the feedback! Your suggestion does make the code more maintainable. But one question. What do you think should happen if the same element is in both 'allowed' and 'denied' sets? Should we wait for Gallaecio's opinion?

Reply:
That is a great question. I think there are two option to consider, and we should think about the intended behavior for such a conflict.

1. If denied should take precedence, then we would want to make sure those elements are excluded regardless of their presence in denied.
2. If the opposite is true and allowed should take precedence, then we should out any elements in denied before considering the allowed set.

I think it would make sense to wait for Gallaecio's opinion so we can fully understand the intended behavior, as this would impact how these conflicts are handled. My guess would be option 1 is preferred, because my instinct is denied should take precedence.

Option 1: denied takes precedence, it must be in allowed and not in denied (AND statement)
```
  def create_checker(allowed, denied):
      if allowed is True:
          return lambda x: x not in denied
      if denied is True:
          return lambda x: x in allowed
      # Denied takes precedence when both sets are provided
      return lambda x: x in allowed and x not in denied
```

Option 2: allowed takes precedence, it is still allowed if in both lists (OR statement)
```
  def create_checker(allowed, denied):
        if allowed is True:
            return lambda x: x not in denied
        if denied is True:
            return lambda x: x in allowed
      return lambda x: x in allowed or x not in denied
```

Reply:
Yeah it really depends on preference. We should wait for the response and discuss it, but thanks for all that interest, I appreciate it.